java/arcs/core/util/performance/PerformanceStatistics.kt

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.util.performance
+
+import arcs.core.util.RunningStatistics
+import arcs.core.util.guardWith
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Utility for tracking running performance/runtime statistics of arbitrary operations.
+ *
+ * Example usage:
```

Comment:
Might be better to just link to the implementation in the test. It's less likely to become out of date